libs/langchain/langchain/llms/nlpcloud.py

```diff
@@ -46,14 +44,8 @@ class NLPCloud(LLM):
     """The number of highest probability tokens to keep for top-k filtering."""
     repetition_penalty: float = 1.0
     """Penalizes repeated tokens. 1.0 means no penalty."""
-    length_penalty: float = 1.0
-    """Exponential penalty to the length."""
-    do_sample: bool = True
-    """Whether to use sampling (True) or greedy decoding."""
     num_beams: int = 1
     """Number of beams for beam search."""
-    early_stopping: bool = False
```

Comment:
note: removing these is a breaking change. if you wanted you could add a @root_validator(pre=True) check that removes deprecated values and provides a deprecation warning without erroring. will let you decide what's best here

Reply:
Thank you @baskaryan that's an interesting suggestion I was not aware of. I'll keep it in mind for next time.
For this one a breaking change is ok because there is the same breaking change in the corresponding nlpcloud Python package that is required by Langchain.
Thanks again!